packages/helix-shared-config/src/IndexConfig.js

```diff
@@ -63,6 +63,31 @@ class IndexConfig extends SchemaDerivedConfig {
     this._document = null;
   }
 
+  /**
+   * Replaces an index definition.
+   *
+   * @param {Object} index index configuration
+   * @param {string} index.name index name
+   * @param {Array} index.include paths to include
+   * @param {Array} index.exclude paths to exclude
+   * @param {string} index.target target
+   * @param {object} index.properties properties to add to the index
+   */
+  replaceIndex({
+    name, include, exclude, target, properties,
+  }) {
```

Comment:
Looks very similar to `addIndex` - not sure whether it'd be more intuitive to add a `addOrReplace` method, although I prefer to block adding an index with an existing name.